CAT-1869 DragNDrop for Sprites, Looks and Sounds

loibi93 opened the pull request:
Context Menus are removed (also Tests are fixed to only use ActionMode when needed). DragNDrop is not implemented to be used in Backpack (since it is not needed there, also confirmed by UX Team).

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "ok to test" to start the testrun.

loibi93 commented:
ok to test

jonny911 commented on catroid/src/org/catrobat/catroid/ui/DynamicListView.java:
```diff
@@ -0,0 +1,528 @@
+
+	public List dataList;
+	private boolean forSpriteList = false;
+
```
Is there a reason for the blank lines?

jonny911 commented on catroidTest/src/org/catrobat/catroid/uitest/ui/fragment/LookFragmentTest.java:
```diff
@@ -819,84 +812,52 @@ public void testRenameLookContextMenu() {
-	}
+		assertEquals("Wrong List before DragAndDropTest", lookDataList.get(0).getLookName(), FIRST_TEST_LOOK_NAME);
+		assertEquals("Wrong List before DragAndDropTest", lookDataList.get(1).getLookName(), SECOND_TEST_LOOK_NAME);
+		assertEquals("Wrong List before DragAndDropTest", lookDataList.get(2).getLookName(), "TestLook0");
```
Is there a reason, why TestLookx is hardcoded?

loibi93 replied:
Yes, as you can see, when the test starts, i add some Looks in a for loop. The names of these looks are composed with TestLook plus a unique number (1, 2 and 3 in this case). So the names are composed at runtime. Same goes for the same issue in NFCTagFragmentTest.

jonny911 commented on catroidTest/src/org/catrobat/catroid/uitest/ui/fragment/NfcTagFragmentTest.java:
```diff
@@ -832,108 +833,52 @@ public void testSelectAllActionModeButton() {
-	}
+	public void testDragAndDropUp() {
+		for (int i = 0; i < 2; i++) {
+			addNFCTagWithName("TestNFC" + i);
```
Is there a reason, why TestNFCx is hardcoded?

loibi93 replied:
same as above.

jonny911 commented on catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java:
```diff
@@ -1632,7 +1632,7 @@ public static void acceptAndCloseActionMode(Solo solo) {
 
 		solo.clickOnView(doneButton);
-		solo.sleep(200);
+		solo.sleep(400);
```
Can this be replaced by solo.waitForxxx()?

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/DynamicListView.java:
```diff
@@ -0,0 +1,519 @@
+			};
+
+	private BitmapDrawable getAndAddHoverView(View v) {
+
```
Remove newline

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/DynamicListView.java:
```diff
@@ -0,0 +1,519 @@
+						return true;
+					}
+					dismissLongPress = true;
+
```
Remove newline

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/DynamicListView.java:
```diff
@@ -0,0 +1,519 @@
+		}
+
+		if (isBelow || isAbove) {
+
```
Remove newline

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/DynamicListView.java:
```diff
@@ -0,0 +1,519 @@
+	}
+
+	private AbsListView.OnScrollListener scrollListener = new AbsListView.OnScrollListener() {
+
```
Remove newline

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/DynamicListView.java:
```diff
@@ -0,0 +1,519 @@
+			return true;
+		}
+
+		if (hoverViewTop + hoverHeight >= height && (offset + extent) < range) {
```
Use an boolean value with an meaningful name for better readability

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/adapter/LookAdapter.java:
```diff
@@ -33,16 +33,58 @@
+
 	private LookFragment lookFragment;
 
+	HashMap<LookData, Integer> idMap = new HashMap<LookData, Integer>();
```
insert private

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/adapter/NfcTagAdapter.java:
```diff
@@ -32,16 +32,49 @@
+
 	private NfcTagFragment nfcTagFragment;
 
+	HashMap<NfcTagData, Integer> idMap = new HashMap<NfcTagData, Integer>();
```
insert private

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/adapter/SoundAdapter.java:
```diff
@@ -35,16 +35,49 @@
+
 	private SoundFragment soundFragment;
 
+	HashMap<SoundInfo, Integer> idMap = new HashMap<SoundInfo, Integer>();
```
insert private

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/adapter/SpriteAdapter.java:
```diff
@@ -41,17 +43,51 @@
+
 	SpritesListFragment spritesListFragment;
 
+	HashMap<Sprite, Integer> idMap = new HashMap<Sprite, Integer>();
```
insert private

thomasmauerhofer commented on catroid/src/org/catrobat/catroid/ui/controller/SoundController.java:
```diff
@@ -253,6 +254,25 @@ public void onClick(View view) {
 			}
 		}
 		holder.soundFragmentButtonLayout.setOnClickListener(listItemOnClickListener);
+
```
remove newline

loibi93 commented:
@thomasmauerhofer fixed issues ;)

loibi93 commented:
Can this be merged now ? @thmq

loibi93 commented:
ok to test

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "ok to test" to start the testrun.

loibi93 commented:
Rebased and squashed to one commit, approved by UX and code reviewed, should be good to go !

CatroidCheck: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidCheck/217/
CatroidSinglePackageEmulatorTest: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/1803/

p4p4 commented:
@loibi93 how to delete a sprite? is there a new way to do this? thanks! :)
